app/assets/javascripts/moneris_simulator/credit_card.coffee

```diff
@@ -0,0 +1,37 @@
+class MonerisSimulator.CreditCard
+  constructor: (args = {}) ->
+    args = $.extend {}, @defaults(), args
+    @validCreditCards = args.validCreditCards
+    @creditCardNumberInput = args.creditCardNumberInput
+    @setUpMonerisPostback()
+
+  setUpMonerisPostback: () ->
+    $(window).on 'message onmessage', (e) =>
+      window.parent.postMessage @response(), "*"
+
+  response: () ->
+    JSON.stringify { dataKey: @dataKey(), bin: @bin(), responseCode: @responseCode() }
+
+  creditCardNumber: () ->
+    @creditCardNumberInput.val().replace /\D/g, ""
+
+  dataKey: () ->
+    "ot-6mj5lng5b3bnnieo6ccf95p3eap4mjd"
+
+  bin: () ->
+    @creditCardNumber().substring 0, 6
+
+  responseCode: () ->
+    if @validCreditCard() then "1" else "51"
+
+  validCreditCard: () ->
+    @creditCardNumber() in @validCreditCards
+
+  defaults: () ->
+    validCreditCards: ["4502285070000007", "4924190000003750", "4551210032790206", "4924190000000145", "4524000000015080", \
```

Comment:
are any of these AMEX? Do we want to generate tokens for AMEX cards too? I think the moneris iframe does, correct ( and fails when you tried to use the token if you don't support the card type)?